src/core/api/instance_api.cpp

```diff
@@ -42,13 +42,19 @@
 #include "common/new.hpp"
 #include "radio/radio.hpp"
 
+#if !defined(OPENTHREAD_BUILD_DATETIME)
 #ifdef __ANDROID__
 #ifdef OPENTHREAD_ENABLE_ANDROID_NDK
 #include <sys/system_properties.h>
 #else
 #include <cutils/properties.h>
 #endif
+#else
```

Comment:
```suggestion
#else // __ANDROID__
```
For better readability?